### PR TITLE
Revert "Revert "MENU-16: [API][Store] Store API""

### PR DESCRIPTION
**Pull request by lmthongit98**
This reverts commit 1b64d5b982c76bf3476e22917341f8a2b5c1673f.

### CODE REVIEW

File: src/main/java/taplink/network/menu/api/controller/GlobalExceptionController.java [hunk 1]
```diff
@@ -24,4 +28,33 @@ public ResponseEntity<ErrorDetails> handleResourceNotFoundException(ResourceNotF
         return new ResponseEntity<>(errorDetails, HttpStatus.NOT_FOUND);
     }
 
+    @ExceptionHandler(SQLIntegrityConstraintViolationException.class)
+    public ResponseEntity<ErrorDetails> conflictData(Exception ex, WebRequest webRequest) {
+        logger.error("SQLIntegrityConstraintViolationException", ex);
```
Comment (hieulm2k) [inline src/main/java/taplink/network/menu/api/controller/GlobalExceptionController.java:33]: Can we create a class constant for this error message?

File: src/main/java/taplink/network/menu/api/controller/GlobalExceptionController.java [hunk 2]
```diff
@@ -24,4 +28,33 @@ public ResponseEntity<ErrorDetails> handleResourceNotFoundException(ResourceNotF
...
+        ErrorDetails errorDetails = new ErrorDetails(new Date(), ex.getMessage(), webRequest.getDescription(false));
+        return new ResponseEntity<>(errorDetails, HttpStatus.CONFLICT);
+    }
+
+    @ExceptionHandler(HttpRequestMethodNotSupportedException.class)
+    public ResponseEntity<ErrorDetails> methodNotSupportedException(Exception ex, WebRequest webRequest) {
+        logger.error("HttpRequestMethodNotSupportedException", ex);
```
Comment (hieulm2k) [inline src/main/java/taplink/network/menu/api/controller/GlobalExceptionController.java:40]: Can we create a class constant for this error message?

File: src/main/java/taplink/network/menu/api/controller/GlobalExceptionController.java [hunk 3]
```diff
@@ -24,4 +28,33 @@ public ResponseEntity<ErrorDetails> handleResourceNotFoundException(ResourceNotF
...
+    public ResponseEntity<ErrorDetails> badRequestHandler(Exception ex, WebRequest webRequest) {
+        logger.error("BadRequestException", ex);
+        ErrorDetails errorDetails = new ErrorDetails(new Date(), ex.getMessage(), webRequest.getDescription(false));
+        return new ResponseEntity<>(errorDetails, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler({Exception.class, IllegalStateException.class})
+    public ResponseEntity<ErrorDetails> handleUnhandledException(Exception ex, WebRequest webRequest) {
+        logger.error("UnhandledException", ex);
+        String message = "There was some error in processing your request. An exception has been logged";
```
Comment (hieulm2k) [inline src/main/java/taplink/network/menu/api/controller/GlobalExceptionController.java:55]: Can we create a class constant for this error message?

File: src/main/java/taplink/network/menu/api/controller/GlobalExceptionController.java [hunk 4]
```diff
@@ -24,4 +28,33 @@ public ResponseEntity<ErrorDetails> handleResourceNotFoundException(ResourceNotF
...
+    @ExceptionHandler({HttpMessageNotReadableException.class, MethodArgumentNotValidException.class})
+    public ResponseEntity<ErrorDetails> badRequestHandler(Exception ex, WebRequest webRequest) {
+        logger.error("BadRequestException", ex);
+        ErrorDetails errorDetails = new ErrorDetails(new Date(), ex.getMessage(), webRequest.getDescription(false));
+        return new ResponseEntity<>(errorDetails, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler({Exception.class, IllegalStateException.class})
+    public ResponseEntity<ErrorDetails> handleUnhandledException(Exception ex, WebRequest webRequest) {
+        logger.error("UnhandledException", ex);
```
Comment (hieulm2k) [inline src/main/java/taplink/network/menu/api/controller/GlobalExceptionController.java:54]: Can we create a class constant for this error message?

File: src/main/java/taplink/network/menu/api/service/impl/StoreServiceImpl.java [hunk 5]
```diff
@@ -31,5 +41,25 @@ public ResponseDto<StoreResponseDto> searchStores(String searchKey, int pageNo,
...
+    @Override
+    public StoreResponseDto createStore(String storeJson, MultipartFile image) {
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            StoreRequestDto storeRequestDto = objectMapper.readValue(storeJson, StoreRequestDto.class);
+            Store store = objectMapperUtils.convertEntityAndDto(storeRequestDto, Store.class);
+            Store savedStore = storeRepository.save(store);
+            return objectMapperUtils.convertEntityAndDto(savedStore, StoreResponseDto.class);
+        } catch (IOException ex) {
+            logger.error("Getting exception while reading storeJson", ex);
```
Comment (hieulm2k) [inline src/main/java/taplink/network/menu/api/service/impl/StoreServiceImpl.java:53]: Can we create a class constant for this error message?

File: src/main/java/taplink/network/menu/api/service/impl/StoreServiceImpl.java [hunk 6]
```diff
@@ -31,5 +41,25 @@ public ResponseDto<StoreResponseDto> searchStores(String searchKey, int pageNo,
...
+            throw new RuntimeException(ex);
+        }
+    }
+
+    @Override
+    public StoreResponseDto findById(Long id) {
+        Store store = storeRepository.findById(id).orElseThrow(() -> new ResourceNotFoundException("Store not found for id: " + id));
```
Comment (hieulm2k) [inline src/main/java/taplink/network/menu/api/service/impl/StoreServiceImpl.java:60]: Can we create a class constant for this error message?

**Review by hieulm2k — CHANGES_REQUESTED** [review]

File: src/main/java/taplink/network/menu/api/service/impl/StoreServiceImpl.java (hunk 6 above)
Comment (nguyenngocbao) [reply src/main/java/taplink/network/menu/api/service/impl/StoreServiceImpl.java:60]: Create a class, and put them to class

Comment (lmthongit98) [reply src/main/java/taplink/network/menu/api/service/impl/StoreServiceImpl.java:60]: Why do we need to create a constant for this message, This is a message for a specific error, we will not reuse this anywhere

Comment (lmthongit98) [reply src/main/java/taplink/network/menu/api/service/impl/StoreServiceImpl.java:60]: But we can put the generic message like {Entity} + could not be found for {id} into the Exception class. Then we can pass specific entity name and the Id to the constructor when we throw the exception

**Pull request merged**